hw/ip/spi_device/data/spi_device_testplan.hjson

```diff
@@ -200,13 +200,206 @@
+            - Configure payload translation bits.
+            - Check proper payload translation is applied.
+            - Disable payload translation for given command.
+            - Check payload is now passing unchanged.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: cmd_info_slots
+      desc: '''
+            - Configure first 5 slots according to required configuration.
+            - Configure next 6 slots for read commands
+            - Randomize configuration of the remaining 13 cmd info slots.
+            - Issue commands with various opcodes enabled in info slots.
+            - Check proper command propagation.
+            - Disable some cmd info slots.
+            - Check no propagation of disabled commands.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: cmd_read_status
+      desc: '''
+            - Configure proper read status command info slot.
+            - Issue read status command.
+            - Check propagation of read status command.
+            - Initiate response to the read status.
+            - Check proper reception of response.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: cmd_read_jedec
+      desc: '''
+            - Configure proper read jedec command info slot.
+            - Issue read jedec command.
+            - Check propagation of read jedec command.
+            - Initiate response to the read jedec.
+            - Check proper reception of response.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: cmd_read_sfdp
+      desc: '''
+            - Configure proper read sfdp command info slot.
+            - Issue read sfdp command.
+            - Check propagation of read sfdp command.
+            - Initiate response to the read sfdp.
+            - Check proper reception of response.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: cmd_fast_read
+      desc: '''
+            - Configure proper fast read command info slot.
+            - Issue fast read command.
+            - Check propagation of fast read command.
+            - Initiate response to the fast read.
+            - Check proper reception of response.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: flash_cmd_upload
+      desc: '''
+            - Configure flash mode.
+            - Configure cmd info slots.
+            - Set upload to 1 for some of 13 non fixed cmd info slots.
+            - Host should poll busy field status to check if command is done.
+            - Issue next command upload and poll busy status again.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: mailbox_command
+      desc: '''
+            - Configure cmd info slots.
+            - Issue one of predefined read command targeting mailbox space.
+            - Check response to read command.
+            - Check if command is processed internally.
+            - Dont issue command larger than 265 bytes.'''
```

Comment:
nit: dont -> don't

Reply:
265 seems strange. 1/4 k is 256. Sorry, I may make a typo in the notes. 
@eunchan can you confirm if this number is correct.

Reply:
@a-will Wanted to check how the current host issues read commands. Does the host system issue a read command more than 256 B payload?

Reply:
For SPI flash, there aren't any restrictions on the length of a host read, and it could be more than 256 bytes. Generally, it's only write and erase commands that have associated sizes (and time to completion).

Note that a host could continuously read through the mailbox space and into the passthrough space, so for emulating a consistent flash device, we'd probably want to pass through read commands regardless of address (even when they start in the mailbox space), but track the address and select the correct data on the output.

Reply:
What is the conclusion here? Should we remove dedicated space-crossing scenario and make crossing possible in base mailbox scenario?

Reply:
OK the first conclusion: No read size limitation. 

The second part (mailbox boundary crossing): In pasthrough, the HW cant' handle the mux. If the address is mailbox, then always mailbox, if not, always read (from the downstream device). In flash, mailbox and read can cross. @a-will @tjaychen what do you expect in this case? Should HW return garbage data after the address crossing the mailbox space? (e.g. read space -> mailbox: send garbage for mailbox, mailbox -> read: send garbage for read).

Reply:
For mailbox crossing, I'd guess it's acceptable to return data from whatever was the original source. If a large read starts in the external flash, passes through the mailbox, and continues to the external flash, we should still see valid external flash data on both sides of the mailbox.

Reply:
@eunchan i feel like having consistent behavior is probably better.  
So maybe just stick with it the original source is fine? 

What would the garbage be in this case though? Would it just be like an aliased address into mailbox?